core/src/main/java/org/mule/runtime/core/internal/store/SharedPartitionedPersistentObjectStore.java

```diff
@@ -21,16 +25,41 @@
  */
 public class SharedPartitionedPersistentObjectStore<T extends Serializable> extends PartitionedPersistentObjectStore<T> {
 
+  public static final String SHARED_PERSISTENT_OBJECT_STORE_KEY = "_defaultSharedPersistentObjectStore";
+
   private File workingDirectory;
+  private Lock lock;
 
   /**
    * Creates a shared partitioned persistent object store.
    *
    * @param workingDirectory {@link File} where to store this OS data. Not null.
+   * @param lockFactory {@link LockFactory} an external lock factory to synchronize the access to this partitioned persistent object store.
    */
-  public SharedPartitionedPersistentObjectStore(File workingDirectory) {
+  public SharedPartitionedPersistentObjectStore(File workingDirectory, LockFactory lockFactory) {
     checkArgument(workingDirectory != null, "workingDirectory cannot be null");
     this.workingDirectory = workingDirectory;
+    this.lock = lockFactory.createLock(getWorkingDirectory());
+  }
+
+  @Override
+  public void open() throws ObjectStoreException {
+    lock.lock();
```

Comment:
use `ConcurrencyUtils#withLock`

Reply:
Doesn't work for me as I have to keep the current interface and throw a checked exception ObjectStoreException and the utility withLock will wrap it with a ReactorException.